engine/helpers.go

```diff
@@ -782,6 +782,12 @@ func (bot *Engine) GetAllExchangeCryptocurrencyDepositAddresses() map[string]map
 							err)
 						continue
 					}
+					if depositAddr == nil {
+						log.Errorf(log.Global, "%s failed to get cryptocurrency deposit address for %s. Err: Deposit address came back as nil.\n",
+							exchName,
+							cryptocurrency)
+						continue
+					}
```

Comment:
I'm not sure we should be adding this section. While the panic was unfortunate, it did highlight a bug in our code, where this would just hide it unless you are looking over logs. Typically for functions across our codebase that have pointer responses, we rely on errors to highlight any issues, including if `nil` is a problem.

I'm not totally against this addition though, but if you are to keep it, you'll need to add it to the two calls to `GetDepositAddress` above

Reply:
I eyeballed our implementations across wrapper files and we haven't done any other mixing from what I can see. I will remove this check.